### PR TITLE
Release notes for in-memory cache subsystem

### PR DESCRIPTION
**Pull request by nerpaula**
Upstreams:

- ~~3.11: https://github.com/arangodb/arangodb/pull/19488~~ docs PR to be re-opened in docs-hugo repo!
  Internal task: https://arangodb.atlassian.net/browse/DOC-555
- devel: https://github.com/arangodb/arangodb/pull/19465

### CODE REVIEW

File: 3.11/release-notes-new-features311.md [hunk 1]
```diff
@@ -982,20 +982,45 @@ Higher values typically mean less compression but faster speeds.
...
+The following startup options have been added:
+
+- `--cache.max-spare-memory-usage`: the maximum memory usage for spare tables
+  in the in-memory cache.
+- `--cache.high-water-multiplier`: controls the cache's effective memory usage
+  limit. The user-defined memory limit (i.e. `--cache.size`) is multiplied with
+  this value to create the effective memory limit, from which on the cache tries
+  to free up memory by evicting the oldest entries.
+
+---
```
Comment (Simran-B) [inline 3.11/release-notes-new-features311.md:1012]: Not a fan of having two sections (above and below this horizontal ruler) with the same version remark. and one about startup options and one about metrics. We have a headline for startup options, but this is rather for the case of options getting added with no clear feature associated with them. If there is a feature with related options and metrics, I would place both under a headline like here, describe the feature briefly, then mention both the startup options and the metrics, with only one version remark under the headline and no horizontal ruler.

File: 3.12/release-notes-new-features312.md [hunk 2]
```diff
@@ -59,20 +59,45 @@ Higher values typically mean less compression but faster speeds.
...
   whether compression is used or not.
 - `rocksdb_cache_edge_compression_ratio`: returns the effective
   compression ratio for all edge cache entries ever stored in the cache.
 
 Note that these metrics are increased upon every insertion into the edge
 cache, but not decreased when data gets evicted from the cache.
 
+### In-memory edge cache startup options and metrics
+
+<small>Introduced in: v3.11.3, 3.12.0</small>
```
Comment (Simran-B) [inline 3.12/release-notes-new-features312.md:78]:
```suggestion
<small>Introduced in: v3.11.3, v3.12.0</small>
```

**Review by Simran-B — CHANGES_REQUESTED** [review]

File: 3.11/release-notes-new-features311.md [hunk 3]
```diff
@@ -982,20 +982,45 @@ Higher values typically mean less compression but faster speeds.
 The following new metrics can be used to determine the usefulness of
 compression:
   
-- `rocksdb_cache_edge_effective_entries_size`: returns the total number of
-  bytes of all entries that were stored in the in-memory edge cache, after
-  compression was attempted/applied. This metric is populated regardless
+- `rocksdb_cache_edge_inserts_effective_entries_size_total`: returns the total
+  number of bytes of all entries that were stored in the in-memory edge cache,
```
[the remark this answers is not in the document]
Comment (nerpaula) [reply 3.11/release-notes-new-features311.md]: Done.

File: 3.11/release-notes-new-features311.md (hunk 1 above)
Comment (nerpaula) [reply 3.11/release-notes-new-features311.md]: Moved it under Miscellaneous changes, kept only one version remark and removed the horizontal ruler.

**Review by Simran-B — APPROVED** [review]

File: 3.12/release-notes-new-features312.md [hunk 4]
```diff
@@ -102,6 +102,29 @@ attempt to create an additional database fails with error
 if other databases are dropped first. The default value for this option is
 unlimited, so an arbitrary amount of databases can be created.
 
+## Miscellaneous changes
+
+### In-memory edge cache startup options and metrics
+
+<small>Introduced in: v3.11.3, v3.12.0</small>
```
Comment (Simran-B) [inline 3.12/release-notes-new-features312.md:109]: Needs to be adjusted!

**Pull request merged**